src/note.ts

```diff
@@ -599,4 +600,60 @@ export abstract class Note extends Tickable {
   setPreFormatted(value: boolean): void {
     this.preFormatted = value;
   }
+
+  // Get the number of beams for this duration
+  getBeamCount(): number {
```

Comment:
Not a fan of these changes -- they pollute `Note` which is already quite polluted. :-)

Reply:
agree